### PR TITLE
Add code quality review policy

### PR DESCRIPTION
**Pull request by jryans**
In the past, we have occasionally written different kinds of tests for Element and the SDKs, but it hasn't been a consistent focus. Going forward, we'd like to change that.

This adds a new review policy which encourages tests for new features. As with everything, we'll continue to tune this based on feedback.

### CODE REVIEW

File: docs/review.md [hunk 1]
```diff
@@ -58,6 +58,35 @@ When reviewing code, here are some things we look for and also things we avoid:
...
+The above policy is not a strict rule, but instead it's meant to be a
+conversation between the author and reviewer. As an author, try to think about
+writing a test when making your next change. As a reviewer, try to think about
+how you might test the area of code you are reviewing. If the reviewer agrees
+it would be quite difficult to test some new feature, then it's okay for them to
+accept the change without tests for now, but we'd eventually like to be more
+strict about this further down the road.
+
+If you do spot areas that are quite hard to test today, please let us know in
+#element-dev:matrix.org. We can work on improving the app architecture and
```
Comment (turt2live) [inline docs/review.md:81]: may want to link this so the parser doesn't think it's a header

File: docs/review.md [hunk 2]
```diff
@@ -58,6 +58,35 @@ When reviewing code, here are some things we look for and also things we avoid:
...
+Element and the SDKs, but it hasn't been a consistent focus. Going forward, we'd
+like to change that.
+
+* For new features, code reviewers will expect some form of automated testing to
+  be included by default
+* For bug fixes, regression tests are of course great to have, but we don't want
+  to block fixes on this, so we won't require them at this time
+
+The above policy is not a strict rule, but instead it's meant to be a
+conversation between the author and reviewer. As an author, try to think about
```
Comment (turt2live) [inline docs/review.md:73]: for labs features, are we requiring (or strongly recommending) tests for each incremental PR or just as a requirement for out-of-labs launch? Concern being that labs features tend to change rapidly during their development, and changing tests all the time sounds overly time consuming. Once the feature is (about to be) stable though, writing tests should not be all that difficult given the incremental reviews.

Existing labs features might have missed the opportunity for tests at launch being a requirement, but new (future) labs features should be testable at each incremental PR, hopefully. Difference might just be that a labs feature would potentially want to introduce tests later if a particular part of the feature is undecided or being live-tested.

Comment (jryans) [reply docs/review.md:73]: Hmm, good question. 🙂 It's a spectrum for sure. I think at the moment, we would lean towards prioritising our ability to evolve the app via labs features and thus not require tests to appear at the same time as the initial cut of labs feature work, as long as (for new labs features going forward at least) the feature author understands that they are effectively deferring part of their work (adding tests) until later and tests are expected to appear before the feature can ship.

**Review by dbkr — APPROVED** [review]

File: docs/review.md (hunk 2 above)
Comment (jryans) [reply docs/review.md:73]: I have folded this bit into the text.

**Pull request merged**